### PR TITLE
Add worldtube observation action

### PR DESCRIPTION
**Pull request by nikwit**
## Proposed changes
Adds an action that observes the coefficients of the  Taylor series of the regular field held by the worldtube singleton.

### CODE REVIEW

File: tests/InputFiles/CurvedScalarWave/WorldtubeKerrSchild.yaml [hunk 1]
```diff
@@ -97,9 +97,9 @@ EventsAndTriggers:
           Offset: 0
     - - PsiAlongAxis1
       - PsiAlongAxis2
-  - - Times:
-        Specified:
-          Values: &FinalTime [2000.]
+  - - TimeCompares:
+        Comparison: GreaterThan
+        Value: 2000.
```
Comment (wthrowe) [inline tests/InputFiles/CurvedScalarWave/WorldtubeKerrSchild.yaml:102]: I am concerned if this didn't work, and I am even more concerned if the new version does.  (For the latter, because you removed the definition of `FinalTime`.)

What was the reason for this change?

File: src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp [hunk 2]
```diff
@@ -155,6 +169,20 @@ struct ExcisionSphere : db::SimpleTag {
   }
 };
 
+/*!
+ * \brief Triggers at which to write the coefficients of the worldtube's
+ * internal Taylor series to file.
+ */
+struct ObserveCoefficientsTrigger : db::SimpleTag {
```
Comment (wthrowe) [inline src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp:176]: Test the tag.

File: tests/InputFiles/CurvedScalarWave/WorldtubeKerrSchild.yaml [hunk 3]
```diff
@@ -109,24 +109,15 @@ EventsAndTriggers:
           StepChoosers:
             - Cfl:
                 SafetyFactor: 0.35
-  - - TimeCompares:
```
Comment (wthrowe) [inline tests/InputFiles/CurvedScalarWave/WorldtubeKerrSchild.yaml:112]: This removal seems unrelated to the rest of the commit.

File: src/Evolution/Systems/CurvedScalarWave/Worldtube/SingletonActions/ObserveWorldtubeSolution.hpp [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+      if (expansion_order > 1) {
+        const auto& psi_quadrupole = db::get<
+            Stf::Tags::StfTensor<Tags::PsiWorldtube, 2, Dim, Frame::Grid>>(box);
+        const auto& dt_psi_quadrupole =
+            db::get<Stf::Tags::StfTensor<::Tags::dt<Tags::PsiWorldtube>, 2, Dim,
+                                         Frame::Grid>>(box);
+        const double wt_radius =
+            db::get<Tags::ExcisionSphere<Dim>>(box).radius();
+        const double trace_psi_2 =
+            3. * (get(psi_monopole) - get(psi_0).at(0)) / wt_radius / wt_radius;
```
Comment (wthrowe) [inline src/Evolution/Systems/CurvedScalarWave/Worldtube/SingletonActions/ObserveWorldtubeSolution.hpp]: `/ square(wt_radius)` and again two lines later.

File: src/Evolution/Systems/CurvedScalarWave/Worldtube/SingletonActions/ObserveWorldtubeSolution.hpp [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+  static Parallel::iterable_action_return_t apply(
+      db::DataBox<DbTagsList>& box,
+      const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+      Parallel::GlobalCache<Metavariables>& cache,
+      const ArrayIndex& /*array_index*/, const ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/) {
+    if (db::get<Tags::ObserveCoefficientsTrigger>(box).is_triggered(box)) {
+      const size_t expansion_order = db::get<Tags::ExpansionOrder>(box);
+      const auto& psi_monopole = db::get<
+          Stf::Tags::StfTensor<Tags::PsiWorldtube, 0, Dim, Frame::Grid>>(box);
```
Comment (wthrowe) [inline src/Evolution/Systems/CurvedScalarWave/Worldtube/SingletonActions/ObserveWorldtubeSolution.hpp:53]: (arbitrary line) I don't know this system well, but there's a lot of math here that is not obviously specific to this observation.  Are there other places where it might be useful, so that breaking it out into a separate function for reuse would make sense?  Or is this all just organizing into the structures required by the observer code, in which case no reason to factor it out.

File: tests/Unit/Evolution/Systems/CurvedScalarWave/Worldtube/SingletonActions/Test_ObserveWorldtubeSolution.cpp [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+      : tt::ConformsTo<Options::protocols::FactoryCreation> {
+    using factory_classes =
+        tmpl::map<tmpl::pair<Trigger, tmpl::list<Triggers::TimeCompares>>>;
+  };
+};
+
+template <typename Generator>
+void check_observe_worldtube_solution(
+    gsl::not_null<Generator*> generator,
+    gsl::not_null<std::uniform_real_distribution<>*> dist,
```
Comment (wthrowe) [inline tests/Unit/Evolution/Systems/CurvedScalarWave/Worldtube/SingletonActions/Test_ObserveWorldtubeSolution.cpp]: `const` these two.

**Review by wthrowe — CHANGES_REQUESTED** [review]

File: tests/InputFiles/CurvedScalarWave/WorldtubeKerrSchild.yaml (hunk 1 above)
Comment (nikwit) [reply tests/InputFiles/CurvedScalarWave/WorldtubeKerrSchild.yaml:102]: I put the removal of the `StepToTimes` in the wrong commit  but fixed it now. 

This always worked, the reason is just to simplify things. The system reaches a steady-state solution so we do not care about a point landing, it just needs to roughly complete at some time.

File: src/Evolution/Systems/CurvedScalarWave/Worldtube/SingletonActions/ObserveWorldtubeSolution.hpp (hunk 5 above)
Comment (nikwit) [reply src/Evolution/Systems/CurvedScalarWave/Worldtube/SingletonActions/ObserveWorldtubeSolution.hpp:53]: 95 % of it is just copying it over into the observer structure so there is not too much point in putting it into free functions.
I am planning on writing more extensive documentation that references the equations and sections of the paper once we publish (hopefully next week)

**Comment by wthrowe** [comment]
OK.  That addresses everything.  Go ahead and squash.

**Review by wthrowe — APPROVED** [review]
